### ISSUE TITLE
Training ESRGAN on 0.5m HR satellite imagery 

### ISSUE THREAD

**Issue by shah0nawaz** [xinntao/BasicSR#204]
**I am training ESRGAN model on satellite imagery, The HR images are 0.5m. I have 20,000 480x480 RGB images. I am training it for scale 4. My training data is not diverse, because I have cropped these images from just 8 large satellite images.  

My Configuration file and some of the results of the model are given in quotation.
First is that the PSNR values does not improve as show in result.

- [ 1] it it possible to train this model on 0.5m resolution satellite imagery to achieve good 
        results.
- [2]  Dataset size is 20,000 480x480 size images.
- Note: I have changed strict = False in def load_network(self, load_path, network, strict=False):
  on strict =  True it gives me an error.

In the initial iterations the validation images results are comparatively better, means images are in some how comparable to the LR images, but latter the there is no structures in the images.  somewhat like smooth plane images with no structures. 
 

Please guide me what I am doing wrong with this setting.**

`19-08-08 14:09:14.035 - INFO:   path:[
    experiments_root: /home/eng.shahnawaz/khan/SR/BasicSR-master/experiments/002_RRDB_ESRGAN_x4_DIV2K
    models: /home/eng.shahnawaz/khan/SR/BasicSR-master/experiments/002_RRDB_ESRGAN_x4_DIV2K/models
    log: /home/eng.shahnawaz/khan/SR/BasicSR-master/experiments/002_RRDB_ESRGAN_x4_DIV2K
    val_images: /home/eng.shahnawaz/khan/SR/BasicSR-master/experiments/002_RRDB_ESRGAN_x4_DIV2K/val_images
    root: /home/eng.shahnawaz/khan/SR/BasicSR-master
    pretrain_model_G: ../experiments/pretrained_models/RRDB_PSNR_x4.pth
    training_state: /home/eng.shahnawaz/khan/SR/BasicSR-master/experiments/002_RRDB_ESRGAN_x4_DIV2K/training_state
  ]
  name: 002_RRDB_ESRGAN_x4_DIV2K
  model: srragan
  use_tb_logger: True
  gpu_ids: [0]
  train:[
    weight_decay_G: 0
    lr_D: 0.0001
    feature_weight: 1
    niter: 500000.0
    lr_G: 0.0001
    beta1_G: 0.9
    gan_weight: 0.005
    lr_gamma: 0.5
    gan_type: vanilla
    lr_steps: [50000, 100000, 200000, 300000]
    lr_scheme: MultiStepLR
    manual_seed: 0
    weight_decay_D: 0
    feature_criterion: l1
    val_freq: 1000.0
    pixel_weight: 0.01
    pixel_criterion: l1
    beta1_D: 0.9
  ]
  datasets:[
    train:[
      mode: LRHR
      use_rot: True
      name: Train
      dataroot_HR: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Train/Train_HR.lmdb
      HR_size: 128
      data_type: lmdb
      dataroot_LR: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Train/Train_LR.lmdb
      batch_size: 16
      use_flip: True
      use_shuffle: True
      phase: train
      n_workers: 8
      scale: 4
    ]
    val:[
      mode: LRHR
      data_type: img
      name: Test
      dataroot_HR: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Test/Test_HR
      phase: val
      dataroot_LR: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Test/Test_LR
      scale: 4
    ]
  ]
  is_train: True
  scale: 4
  network_D:[
    mode: CNA
    norm_type: batch
    in_nc: 3
    which_model_D: discriminator_vgg_128
    act_type: leakyrelu
    nf: 64
  ]
  logger:[
    print_freq: 200
    save_checkpoint_freq: 5000.0
  ]
  network_G:[
    mode: CNA
    in_nc: 3
    nb: 23
    group: 1
    nf: 64
    out_nc: 3
    norm_type: None
    gc: 32
    which_model_G: RRDB_net
    scale: 4
  ]

19-08-08 14:09:14.117 - INFO: Random seed: 0
19-08-08 14:09:14.119 - INFO: Read lmdb keys from cache: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Train/Train_HR.lmdb/_keys_cache.p
19-08-08 14:09:14.125 - INFO: Read lmdb keys from cache: /home/eng.shahnawaz/Khan/SR/BasicSR-master/codes/data/Train/Train_LR.lmdb/_keys_cache.p
19-08-08 14:09:14.131 - INFO: Dataset [LRHRDataset - Train] is created.
19-08-08 14:09:14.131 - INFO: Number of train images: 19,749, iters: 1,235
19-08-08 14:09:14.132 - INFO: Total epochs needed: 405 for iters 500,000
19-08-08 14:09:14.132 - INFO: Dataset [LRHRDataset - Test] is created.
19-08-08 14:09:14.132 - INFO: Number of val images in [Test]: 25
19-08-08 14:09:14.335 - INFO: Initialization method [kaiming]
19-08-08 14:09:17.736 - INFO: Initialization method [kaiming]
19-08-08 14:09:17.879 - INFO: Loading pretrained model for G [../experiments/pretrained_models/RRDB_PSNR_x4.pth] ...
19-08-08 14:09:19.637 - INFO: Network G structure: DataParallel - RRDBNet, with parameters: 1
19-08-08 14:09:19.657 - INFO: Model [SRRaGANModel] is created.
19-08-08 14:09:19.657 - INFO: Start training from epoch: 0, iter: 0
19-08-08 14:11:23.659 - INFO: <epoch:  0, iter:     200, lr:1.000e-04> l_g_pix: 1.5828e-03 l_g_fea: 1.1327e+00 l_g_gan: 2.3663e-02 l_d_real: 4.1374e-02 l_d_fake: 3.6769e-02 D_real: -5.4052e+00 D_fake: -1.0099e+01 
19-08-08 14:13:25.695 - INFO: <epoch:  0, iter:     400, lr:1.000e-04> l_g_pix: 1.6721e-03 l_g_fea: 1.1168e+00 l_g_gan: 3.3311e-02 l_d_real: 2.2935e-02 l_d_fake: 1.7558e-02 D_real: 1.9460e+00 D_fake: -4.6960e+00 
19-08-08 14:15:27.492 - INFO: <epoch:  0, iter:     600, lr:1.000e-04> l_g_pix: 1.4723e-03 l_g_fea: 9.6328e-01 l_g_gan: 2.7070e-02 l_d_real: 7.5555e-02 l_d_fake: 1.6161e-01 D_real: -5.5921e+00 D_fake: -1.0888e+01 
19-08-08 14:17:28.620 - INFO: <epoch:  0, iter:     800, lr:1.000e-04> l_g_pix: 1.9521e-03 l_g_fea: 1.1159e+00 l_g_gan: 3.7583e-02 l_d_real: 1.3740e-03 l_d_fake: 1.3976e-03 D_real: -1.2553e+01 D_fake: -2.0068e+01 
19-08-08 14:19:29.849 - INFO: <epoch:  0, iter:   1,000, lr:1.000e-04> l_g_pix: 1.4149e-03 l_g_fea: 9.8995e-01 l_g_gan: 1.1891e-02 l_d_real: 2.1871e-01 l_d_fake: 2.3837e-01 D_real: -8.4759e+00 D_fake: -1.0625e+01 
19-08-08 14:19:33.284 - INFO: # Validation # PSNR: 1.2244e+01
19-08-08 14:19:33.284 - INFO: <epoch:  0, iter:   1,000> psnr: 1.2244e+01
19-08-08 14:21:35.022 - INFO: <epoch:  0, iter:   1,200, lr:1.000e-04> l_g_pix: 1.6472e-03 l_g_fea: 1.0651e+00 l_g_gan: 4.8405e-02 l_d_real: 2.8368e-04 l_d_fake: 1.5357e-04 D_real: -1.6155e+01 D_fake: -2.5836e+01 
19-08-08 14:23:36.950 - INFO: <epoch:  1, iter:   1,400, lr:1.000e-04> l_g_pix: 1.3944e-03 l_g_fea: 1.0673e+00 l_g_gan: 7.0048e-03 l_d_real: 7.7331e-01 l_d_fake: 5.8031e-01 D_real: -1.3569e+01 D_fake: -1.4293e+01 
19-08-08 14:25:38.254 - INFO: <epoch:  1, iter:   1,600, lr:1.000e-04> l_g_pix: 1.5914e-03 l_g_fea: 9.2307e-01 l_g_gan: 6.0770e-02 l_d_real: 5.4782e-05 l_d_fake: 1.7941e-05 D_real: 8.7478e-01 D_fake: -1.1279e+01 
19-08-08 14:27:39.592 - INFO: <epoch:  1, iter:   1,800, lr:1.000e-04> l_g_pix: 2.3072e-03 l_g_fea: 1.1331e+00 l_g_gan: 3.4904e-02 l_d_real: 1.6022e-03 l_d_fake: 3.8206e-03 D_real: 4.3368e+00 D_fake: -2.6413e+00 
19-08-08 14:29:40.935 - INFO: <epoch:  1, iter:   2,000, lr:1.000e-04> l_g_pix: 1.8213e-03 l_g_fea: 1.1527e+00 l_g_gan: 1.4508e-02 l_d_real: 1.0136e-01 l_d_fake: 8.3013e-02 D_real: -1.0317e+01 D_fake: -1.3127e+01 
19-08-08 14:29:44.341 - INFO: # Validation # PSNR: 1.3224e+01
19-08-08 14:29:44.341 - INFO: <epoch:  1, iter:   2,000> psnr: 1.3224e+01
19-08-08 14:31:46.300 - INFO: <epoch:  1, iter:   2,200, lr:1.000e-04> l_g_pix: 1.5068e-03 l_g_fea: 9.9740e-01 l_g_gan: 5.0837e-02 l_d_real: 2.6262e-03 l_d_fake: 1.1942e-04 D_real: 1.0651e+01 D_fake: 4.8495e-01 
19-08-08 14:33:48.238 - INFO: <epoch:  1, iter:   2,400, lr:1.000e-04> l_g_pix: 1.7282e-03 l_g_fea: 1.1726e+00 l_g_gan: 1.3577e-02 l_d_real: 1.4982e-01 l_d_fake: 1.3480e-01 D_real: -3.6998e+00 D_fake: -6.2729e+00 
19-08-08 14:35:50.629 - INFO: <epoch:  2, iter:   2,600, lr:1.000e-04> l_g_pix: 1.5493e-03 l_g_fea: 1.2311e+00 l_g_gan: 4.9144e-02 l_d_real: 1.0645e-04 l_d_fake: 2.0470e-04 D_real: -1.2386e+01 D_fake: -2.2215e+01 
19-08-08 14:37:52.585 - INFO: <epoch:  2, iter:   2,800, lr:1.000e-04> l_g_pix: 1.3867e-03 l_g_fea: 9.9951e-01 l_g_gan: 1.0637e-02 l_d_real: 1.9037e-01 l_d_fake: 1.7257e-01 D_real: -1.3567e+01 D_fake: -1.5513e+01 
19-08-08 14:39:54.554 - INFO: <epoch:  2, iter:   3,000, lr:1.000e-04> l_g_pix: 1.5156e-03 l_g_fea: 
19-08-09 08:13:14.066 - INFO: # Validation # PSNR: 1.4323e+01
.
.
.
.
.
.
19-08-09 08:13:14.066 - INFO: <epoch: 85, iter: 106,000> psnr: 1.4323e+01
19-08-09 08:15:15.782 - INFO: <epoch: 86, iter: 106,200, lr:2.500e-05> l_g_pix: 1.7283e-03 l_g_fea: 1.1521e+00 l_g_gan: 5.5051e-02 l_d_real: 1.0917e-04 l_d_fake: 1.6008e-04 D_real: -2.6778e+01 D_fake: -3.7788e+01 
19-08-09 08:17:17.356 - INFO: <epoch: 86, iter: 106,400, lr:2.500e-05> l_g_pix: 2.1094e-03 l_g_fea: 1.0364e+00 l_g_gan: 6.5800e-02 l_d_real: 3.8147e-06 l_d_fake: 4.4480e-06 D_real: -3.1164e+01 D_fake: -4.4324e+01 
19-08-09 08:19:18.842 - INFO: <epoch: 86, iter: 106,600, lr:2.500e-05> l_g_pix: 1.5656e-03 l_g_fea: 1.1360e+00 l_g_gan: 6.6045e-02 l_d_real: 4.5746e-06 l_d_fake: 2.2426e-06 D_real: -2.8864e+01 D_fake: -4.2073e+01 
19-08-09 08:21:20.140 - INFO: <epoch: 86, iter: 106,800, lr:2.500e-05> l_g_pix: 1.2812e-03 l_g_fea: 1.1066e+00 l_g_gan: 6.0040e-02 l_d_real: 2.4169e-05 l_d_fake: 7.8603e-06 D_real: -3.1343e+01 D_fake: -4.3351e+01 
19-08-09 08:23:21.394 - INFO: <epoch: 86, iter: 107,000, lr:2.500e-05> l_g_pix: 1.7145e-03 l_g_fea: 1.1083e+00 l_g_gan: 6.4715e-02 l_d_real: 1.2479e-05 l_d_fake: 4.5374e-06 D_real: -2.5421e+01 D_fake: -3.8364e+01 
19-08-09 08:23:24.709 - INFO: # Validation # PSNR: 1.3831e+01
19-08-09 08:23:24.709 - INFO: <epoch: 86, iter: 107,000> psnr: 1.3831e+01
19-08-09 08:25:25.996 - INFO: <epoch: 86, iter: 107,200, lr:2.500e-05> l_g_pix: 1.4819e-03 l_g_fea: 1.0389e+00 l_g_gan: 7.2056e-02 l_d_real: 1.4378e-05 l_d_fake: 4.0903e-06 D_real: -2.5625e+01 D_fake: -4.0036e+01 
19-08-09 08:27:27.746 - INFO: <epoch: 87, iter: 107,400, lr:2.500e-05> l_g_pix: 1.5971e-03 l_g_fea: 9.9606e-01 l_g_gan: 9.4154e-02 l_d_real: 1.4901e-08 l_d_fake: 7.4506e-09 D_real: -2.6405e+01 D_fake: -4.5235e+01 
19-08-09 08:29:29.008 - INFO: <epoch: 87, iter: 107,600, lr:2.500e-05> l_g_pix: 1.9925e-03 l_g_fea: 1.0228e+00 l_g_gan: 7.8156e-02 l_d_real: 4.5449e-07 l_d_fake: 3.7998e-07 D_real: -2.1210e+01 D_fake: -3.6841e+01 
    `
